### ISSUE TITLE
BUG setTotalRecords() not work in version 10.4.0

### ISSUE THREAD

**Issue by rodrigocabraln** [yajra/laravel-datatables#2998]
The documentation says:
"In some cases, we need to manually set the total records of our DataTables and skip its internal counting functionality. To achieve this, we can use setTotalRecords($count) api."

Using setTotalRecords() in version 9.x does not generate the SQL count query, but if it is used in version 10.4.0 the count query is generated.

Also in version 9.x the following SQL is generated for a simple datatables:

```
[2023-05-12 16:29:37] local.INFO: select count(*) as aggregate from `cuentas_bancarias` where `cuentas_bancarias`.`deleted_at` is null  
[2023-05-12 16:29:37] local.INFO: select * from `cuentas_bancarias` where `cuentas_bancarias`.`deleted_at` is null order by `banco` asc limit 10 offset 0  
```

The same code in version 10.4.0 duplicates the statements:

```
[2023-05-12 16:34:19] local.INFO: select count(*) as aggregate from `cuentas_bancarias` where `cuentas_bancarias`.`deleted_at` is null  
[2023-05-12 16:34:19] local.INFO: select count(*) as aggregate from `cuentas_bancarias` where `cuentas_bancarias`.`deleted_at` is null  
[2023-05-12 16:34:19] local.INFO: select * from `cuentas_bancarias` where `cuentas_bancarias`.`deleted_at` is null order by `banco` asc limit 10 offset 0  
```

**Comment by rodrigocabraln** [comment]
Please solve this bug, it overloads the server

**Comment by rodrigocabraln** [comment]
10.4.1 does not resolve the issue

**Comment by yajra** [comment]
> The same code in version 10.4.0 duplicates the statements:

Expected behavior on initial load of dataTables:

1st query - total count
2nd query - filtered count
3rd query - paginated data

However, some optimization was added on previous version which skips filtered count on initial load but it introduces some issues on other features of the package. 

Don't have time to check this further yet. Please do not hesitate to submit a PR if you can fix. Thanks!

**Comment by yajra** [comment]
Please try again on v10.4.3, fixed via #3007
